openpype/hosts/maya/api/plugin.py

```diff
@@ -295,6 +302,10 @@ def update_instances(self, update_list):
     def remove_instances(self, instances):
         return self._default_remove_instances(instances)
 
+    def apply_settings(self, project_settings, system_settings):
+        """Method called on initialization of plugin to apply settings."""
+        apply_settings(self, project_settings, system_settings)
```

Comment:
Can be removed because it's also on `MayaCreatorBase` right?

Reply:
Its not on `MayaCreatorBase` though?

Its on `MayaCreator`.

Reply:
I think I got confused by how Github was hiding unchanged lines - sorry about that.

Anyway, what's the reason we wouldn't want it on the base? :) Just because that's also used for the legacy convertor?

---

@iLLiCiTiT what if we create a mixin class to serve this functionality for all hosts?

```python
CreatorAutoApplySettingsMixin(object):

    settings_name = None

    def apply_settings(self, project_settings, system_settings):
        """Method called on initialization of plugin to apply settings."""
        apply_settings(self, project_settings, system_settings)
```

Only tricky thing with that mixin is that'd need to appear BEFORE the Creator in the inheritance:

```python
class MyCreator(Mixin, Creator):
    # works, because Mixin.apply_settings overrides Creator.apply_settings
    pass

class MyCreator(Creator, Mixin):
    # doesn't work, because Creator.apply_settings overrides Mixin.apply_settings
    pass
```

Reply:
Discussion for a different place and time. We'll be changing few things in creators regarding settings, chasing mixins would be another hell. Keep it as it is now, please.